### PR TITLE
add support for ignoring branches

### PR DESCRIPTION
**Pull request by PixelJonas**
Hey guys,

this PR adds the capability to "exclude branches" from the WebHook. I used your "committersToIgnore" as an example.

I moved some of the logic validating the Event to it's own JAVA class. You can find the logic at `src/main/java/nl/topicus/bitbucket/api/PushEventService.java`

I think using RegEx for branches to be excluded is the best way. So we'd be able to do stuff like "release/.*"

If the configuration paramter is missing or empty, everything should be "valid".

I hope you accept this PR.

Best Regards,
Jonas

### CODE REVIEW

**Comment by PixelJonas** [comment]
So is there any update if this going to be merged?

**Comment by casz** [comment]
@PixelJonas released https://marketplace.atlassian.com/plugins/nl.topicus.bitbucket.bitbucket-webhooks/versions#b100600000 released :)

**Comment by casz** [comment]
@PixelJonas this has broken PR's from being triggered properly on Jenkins, I just been through long testing phase on productions :|
Looking at the code changes you made, I cannot see where we are going wrong. But I basically downgraded to 1.5.2 and PR events starting being triggered on Jenkins :(

**Comment by casz** [comment]
There is no debug output on Jenkins or Bitbucket logs :| So I am flailing in the dark.

**Comment by dantran** [comment]
jenkins user group also reported bb-webhook 1.6 breaks their PR build

**Comment by PixelJonas** [comment]
I'll try to reproduce/fix this this weekend.

**Comment by casz** [comment]
@PixelJonas I created #77

File: src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java [hunk 1]
```diff
@@ -298,54 +296,36 @@ private void sendEvents(Object event, Repository repo, EventType eventType)
...
                                 webHookConfiguration.getTitle(),
                                 webHookConfiguration.getURL(),
-                                webHookConfiguration.getCommittersToIgnore(),
                                 body);
-                        continue;
                     }
                 }
-            }
-
-            try (CloseableHttpResponse response = httpClient.execute(post))
```
Comment (casz) [inline src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java]: The mistake is here, we I never sent out others events then push events.

File: src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java [hunk 2]
```diff
@@ -298,54 +296,36 @@ private void sendEvents(Object event, Repository repo, EventType eventType)
...
             post.setURI(URI.create(webHookConfiguration.getURL()));
+            PushEventService pushEventService = new PushEventService(webHookConfiguration);
 
-            if(event instanceof BitbucketPushEvent)
+            if(event instanceof BitbucketPushEvent && pushEventService.isValidEvent((BitbucketPushEvent) event, webHookConfiguration))
             {
-                BitbucketPushEvent bitbucketPushEvent = (BitbucketPushEvent) event;
-                if (webHookConfiguration.getCommittersToIgnore() != null &&
-                        bitbucketPushEvent.getActor() != null)
+                try (CloseableHttpResponse response = httpClient.execute(post))
```
Comment (casz) [inline src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java]: This only lets BitbucketPushEvent set events

File: src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java (hunk 1 above)
Comment (PixelJonas) [reply src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java]: I see ... I added some tests to my local instance and will fix this asap. I'll also add some tests to make sure this isn't happening again ;)

**Pull request merged**